Objects/longobject.c

```diff
@@ -64,6 +64,10 @@ get_small_int(sdigit ival)
     do if (-NSMALLNEGINTS <= ival && ival < NSMALLPOSINTS) { \
         return get_small_int((sdigit)ival); \
     } while(0)
+#define CHECK_SMALL_UINT(ival) \
+    do if (ival < NSMALLPOSINTS) { \
```

Comment:
Based on the above conclusions, would it be helpful to at least add a comment, mentioning why the cast is not needed in this case?

```suggestion
    do if (ival < NSMALLPOSINTS) { /* since ival is a literal, a cast is not needed */ \ 
```

Reply:
No, I think best not to. The comment would be more work for the reader than simply adding the cast would :wink: 

The code does exactly the simple thing that it looks like it does. If it did something tricky, or if it might look like it did one thing but it actually did another, that would be a good occasion for a comment.

Reply:
@gnprice Ah okay, fair enough. I mostly just thought of adding it because there seemed to be some degree of nuance involved here. Also, it may help readers understand why there isn't a cast here. But that's a good point, it might be adding to much additional work for the reader.